utils/pattern/src/implementations.rs

```diff
@@ -0,0 +1,32 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{Pattern, SinglePlaceholder, SinglePlaceholderPattern};
+
+#[cfg(feature = "alloc")]
+use alloc::boxed::Box;
+use zerovec::{maps::ZeroMapKV, ule::VarULE, VarZeroSlice, VarZeroVec, ZeroVecError};
+
+#[cfg(feature = "alloc")]
+impl<'a> ZeroMapKV<'a> for Pattern<SinglePlaceholder, str> {
+    type Container = VarZeroVec<'a, Pattern<SinglePlaceholder, str>>;
+    type Slice = VarZeroSlice<Pattern<SinglePlaceholder, str>>;
+    type GetType = Pattern<SinglePlaceholder, str>;
+    type OwnedType = Box<Pattern<SinglePlaceholder, str>>;
+}
+
+#[cfg(feature = "alloc")]
+unsafe impl VarULE for Pattern<SinglePlaceholder, str> {
```

Comment:
Thought: this impl is narrower than required; better would be an impl where Store: VarULE

Non blocking because we can generalize later. But up to this point we've done a good job of keeping the impls general.

Reply:
I agree, but I am thinking also to add imple to Double Pattern or  generalize this later.